src/project/manifest.rs

```diff
@@ -14,8 +17,554 @@ use serde_with::{serde_as, DeserializeAs, DisplayFromStr, PickFirst};
 use std::collections::HashMap;
 use std::ops::Range;
 use std::path::{Path, PathBuf};
+use toml_edit::{Array, Document, Item, Table, TomlError, Value};
 use url::Url;
 
+/// Handles the project's manifest file.
+/// This struct is responsible for reading, parsing, editing, and saving the manifest.
+/// It encapsulates all logic related to the manifest's TOML format and structure.
+/// The manifest data is represented as a [`ProjectManifest`] struct for easy manipulation.
+/// Owned by the [`Project`] struct, which governs its usage.
+///
+#[derive(Debug, Clone)]
+pub struct Manifest {
+    /// The path to the manifest file
+    pub path: PathBuf,
+
+    /// The raw contents of the manifest file
+    pub contents: String,
+
+    /// Editable toml document
+    pub document: toml_edit::Document,
+
+    /// The parsed manifest
+    pub manifest: ProjectManifest,
```

Comment:
should this be called `parsed` or something like this? Might make the distinction more obvious.

Reply:
Good one, let me discuss with gary

Reply:
Renamed it to that!